src/shared/serverinfo.hpp

```diff
@@ -0,0 +1,231 @@
+#ifndef SERVERINFO_HPP
+#define SERVERINFO_HPP
+
+#include <vector>
+#include <string>
+
+#include "defs.hpp"
+
+struct serverinfo;
+
+// list of known servers, with or without informations
+// TODO:
+//   * should be a private static of serverinfo
+//   * stop using raw pointers (RAII ftw)
+extern vector<serverinfo*> servers;
+
+// code imported, here to reduce build dependencies
+extern bool filterstring(char *dst, const char *src, bool newline, bool colour, bool whitespace, bool wsstrip, size_t len);
+namespace client
+{
+    extern bool serverinfocompare(serverinfo const*a, serverinfo const*b);
+}
+
+namespace server
+{
+    extern int getver( int );
+}
+
+//this enum lists (some of) the server properties that can be used for sorting.
+//The C and C++ specs says and enum starts with the value 0 except if explicitly
+//specified.
+//It have 2 list of members:
+// * row 1: original names, for compat, and maybe to allow some naming support in
+//          CS latter
+// * row 2: aliases to avoid reapeating 'SINFO::' for nothing.
+// TODO: use scoped enums
```

Comment:
Grammar is a bit off. Also, I don't think we need to explain C++ standards there.

```suggestion
/**
 * (some of) the server properties that can be used for sorting
 *
 * it has 2 types of members:
 *  - row 1: original names, for compat, and maybe to allow some naming support in CS later
// * row 2: aliases to avoid reapeating 'SINFO::' for nothing.
// TODO: use scoped enums
```

Even though all enums are initialized with `0`, I tend to hardcode the starting value in enums by the way, following the logic "it's not a good idea if you have to explain it". I think that's also done in the remaining codebase.